src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -165,11 +172,82 @@ public void onBreedingGooey(SelectForBreedingEvent event, EntityRef gooeyEntity,
      */
     @ReceiveEvent
     public void onBeginBreedingProcess(BeginBreedingEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent) {
-        Prefab hatchlingPrefab = prefabManager.getPrefab("GooKeeper:gooeyHatchling");
-        EntityBuilder entityBuilder = entityManager.newBuilder(hatchlingPrefab);
-        LocationComponent locationComponent = entityBuilder.getComponent(LocationComponent.class);
-        LocationComponent gooeyLocation = gooeyEntity.getComponent(LocationComponent.class);
-        locationComponent.setWorldPosition(gooeyLocation.getWorldPosition().add(new Vector3f(2f, 1f, 2f)));
-        entityBuilder.build();
+        delayManager.addDelayedAction(gooeyEntity, Constants.spawnGooeyEggEventID, random.nextLong(4000, 7000));
+        AggressiveComponent aggressiveComponent = new AggressiveComponent();
+        FriendlyComponent friendlyComponent = new FriendlyComponent();
+        NeutralComponent neutralComponent = new NeutralComponent();
+
+        MatingComponent matingComponent = gooeyEntity.getComponent(MatingComponent.class);
+
+        AggressiveComponent parent1Aggressive = gooeyEntity.getComponent(AggressiveComponent.class);
+        AggressiveComponent parent2Aggressive = matingComponent.matingWithEntity.getComponent(AggressiveComponent.class);
+
+        if (parent1Aggressive != null) {
+            aggressiveComponent.aggressivenessFactor += parent1Aggressive.aggressivenessFactor;
+        }
+
+        if (parent2Aggressive != null) {
+            aggressiveComponent.aggressivenessFactor += parent2Aggressive.aggressivenessFactor;
+        }
+
+        aggressiveComponent.aggressivenessFactor /= 2f;
```

Comment:
This may be overly complicated, but you've got a decent chunk of repeated code here that could be extracted and simplified.

Consider making an abstract component with a single public field:
```java
public abstract FactorComponent implements Component {
    public float magnitude;
}
```

Then, make `AggressiveComponent`, `NeutralComponent`, and `FriendlyComponent` all extend `FactorComponent`

Next, make a new method that takes a given class adds the two factors together, into a new component. Something like

```java
public <T extends FactorComponent> T getChildFactor(T childComponent, EntityRef parent1, EntityRef parent2) {
    FactorComponent source1 = parent1.getComponent(childComponent.getClass());
    FactorComponent source2 = parent2.getComponent(childComponent.getClass());
    /* Do null checks and add them to `childComponent` */
    /* Divide by scale factor /*
    /* Return child component */
}
```